core/src/main/scala/matryoshka/Nu.scala

```diff
@@ -20,25 +20,28 @@ import Recursive.ops._
 
 import scalaz._, Scalaz._
 
-final case class Nu[F[_]](unNu: Id ~> F, a: Id[_])
+/** This is for coinductive (potentially infinite) recursive structures, models
+  * the concept of “codata”, aka, the “greatest fixed point”.
+  */
+sealed abstract class Nu[F[_]]
+private final case class Nu0[F[_], A](unNu: A => F[A], a: A) extends Nu[F]
```

Comment:
@wemrysi Any arguments for or against either? My only argument for the current one is that it is _pretty_ close to `ana = Nu`, which I think is helpful in understanding the structure (and also happens to be how it’s defined in Haskell).